tests/components/tasmota/test_common.py

```diff
@@ -0,0 +1,328 @@
+"""Common test objects."""
+import copy
+import json
+
+from hatasmota.const import (
+    CONF_MAC,
+    CONF_OFFLINE,
+    CONF_ONLINE,
+    CONF_PREFIX,
+    PREFIX_CMND,
+    PREFIX_TELE,
+)
+from hatasmota.utils import (
+    get_state_offline,
+    get_state_online,
+    get_topic_tele_state,
+    get_topic_tele_will,
+)
+
+from homeassistant.components.tasmota.const import DEFAULT_PREFIX
+from homeassistant.const import STATE_UNAVAILABLE
+
+from tests.async_mock import ANY
+from tests.common import async_fire_mqtt_message
+
+
+async def help_test_availability_when_connection_lost(
+    hass, mqtt_client_mock, mqtt_mock, domain, config
+):
+    """Test availability after MQTT disconnection."""
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{config[CONF_MAC]}/config",
+        json.dumps(config),
+    )
+    await hass.async_block_till_done()
+    async_fire_mqtt_message(
+        hass,
+        get_topic_tele_will(config),
+        get_state_online(config),
+    )
+
+    state = hass.states.get(f"{domain}.test")
+    assert state.state != STATE_UNAVAILABLE
+
+    mqtt_mock.connected = False
+    mqtt_client_mock.on_disconnect(None, None, 0)
```

Comment:
Please run this on the executor since it should run in sync context.

Reply:
Fixed